src/button/button-group.scss

```diff
@@ -6,45 +6,53 @@
   display: inline-flex;
   align-items: center;
 
-  > .iui-button {
+  $childSelector: '&, input, button'; // target nested inputs and buttons
+
+  > * {
     &:not(.iui-borderless) {
-      border-radius: 0;
-    }
+      #{$childSelector} {
+        border-radius: 0;
+      }
+
+      &:first-child {
+        #{$childSelector} {
+          border-top-left-radius: $iui-border-radius;
+          border-bottom-left-radius: $iui-border-radius;
+        }
+      }
 
-    &:first-of-type:not(.iui-borderless) {
-      border-top-left-radius: $iui-border-radius;
-      border-bottom-left-radius: $iui-border-radius;
+      &:last-child {
+        #{$childSelector} {
+          border-top-right-radius: $iui-border-radius;
+          border-bottom-right-radius: $iui-border-radius;
+        }
+      }
     }
 
-    &:last-of-type:not(.iui-borderless) {
-      border-top-right-radius: $iui-border-radius;
-      border-bottom-right-radius: $iui-border-radius;
+    &:not(:last-child) {
+      margin-right: -1px;
     }
 
-    &:not(.iui-borderless):not(.iui-cta):not(.iui-high-visibility):not(:last-of-type) {
-      border-right: none;
+    &:hover,
+    &:focus,
+    &:focus-within {
+      z-index: 2;
```

Comment:
I'm afraid this z-index could cause unexpected issues in some places but I can't think of any

Reply:
@veekeys you seen some `z-index` related bugs in APIM. Any thoughts?

Reply:
Yeah.... Is there any different solution than z-index? I would urge not to use it if possible. Usually z-index issues appear when using multiple z-index based components in single place (in APIM there was with the popover based elements and overlay).

Reply:
As I understand it, the `z-index` was suggested so that the outer glow effect on focus would not get clipped by the element directly to the right of it.  This is a known issue in many other components, but will be fixed when #166 passes.  I vote we remove the `z-index` and approve this PR knowing that the issue is wide spread but will be fixed soon.

Reply:
It will NOT be automatically fixed when #166 passes, but it could maybe be worked around with some nasty border hacks.

> in APIM there was with the popover based elements and overlay

would like to see an example

Reply:
I created a new stacking context using `isolation: isolate` so the z-index should be sealed off from outside influence.